analyzer.go

```diff
@@ -179,6 +250,42 @@ func (a *Analyzer) buildpacks() map[string]struct{} {
 	return buildpacks
 }
 
+func (a *Analyzer) removeOldBackpackLayersNotInGroup(groupBPs map[string]struct{}, launchDir string) error {
+	cachedBPs, err := cachedBuildpacks(launchDir)
+	if err != nil {
+		return err
+	}
+
+	for _, cachedBP := range cachedBPs {
+		_, exists := groupBPs[cachedBP]
+		if !exists {
+			a.Out.Printf("removing cached layers for buildpack '%s' not in group\n", cachedBP)
+			if err := os.RemoveAll(filepath.Join(launchDir, cachedBP)); err != nil && !os.IsNotExist(err) {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func cachedBuildpacks(launchDir string) ([]string, error) {
+	cachedBps := make([]string, 0, 0)
+	bpDirs, err := filepath.Glob(filepath.Join(launchDir, "*"))
+	if err != nil {
+		return nil, err
+	}
+	for _, dir := range bpDirs {
+		info, err := os.Stat(dir)
+		if err != nil {
+			return nil, err
+		}
+		if filepath.Base(dir) != "app" && info.IsDir() {
```

Comment:
Use `os.SameFile` with chosen app dir, as in: https://github.com/buildpack/lifecycle/blob/master/launcher.go